src/components/select/filterable-select/filterable-select.test.js

```diff
@@ -0,0 +1,1291 @@
+    it("should close the list with the Tab key", () => {
+      CypressMountWithProviders(<FilterableSelectComponent />);
+
+      dropdownButton().click();
+      selectList().should("be.visible");
+      selectInput().tab();
+      selectInput().should("have.attr", "aria-expanded", "false");
+      selectList().should("not.exist");
+    });
+
+    it("should close the list with the Esc key", () => {
+      CypressMountWithProviders(<FilterableSelectComponent />);
+
+      dropdownButton().click();
+      selectList().should("be.visible");
+      selectInput().trigger("keydown", { ...keyCode("Esc") });
+      selectInput().should("have.attr", "aria-expanded", "false");
+      selectList().should("not.exist");
+    });
+
+    it("should close the list by clicking out of the component", () => {
+      CypressMountWithProviders(<FilterableSelectComponent />);
+
+      dropdownButton().click();
+      selectList().should("be.visible");
+      body().click({ force: true });
+      selectInput().should("have.attr", "aria-expanded", "false");
+      selectList().should("not.exist");
+    });
+
+    it.each([
+      ["open", "downarrow"],
+      ["open", "uparrow"],
+      ["open", "Home"],
+      ["open", "End"],
+      ["not open", "Enter"],
+    ])(
+      "should %s the list when %s is pressed with Filterable Select input in focus",
+      (state, key) => {
+        CypressMountWithProviders(<FilterableSelectComponent />);
+
+        commonDataElementInputPreview().focus();
+        selectInput().trigger("keydown", { ...keyCode(key), force: true });
+        if (state === "open") {
+          selectList().should("be.visible");
+        } else {
+          selectList().should("not.exist");
+        }
+      }
+    );
+
+    it.each([["Amber"], ["Yellow"]])(
+      "should select option %s when clicked from the list",
+      (option) => {
+        CypressMountWithProviders(<FilterableSelectComponent />);
+
+        dropdownButton().click();
+        selectListText(option).click();
+        getDataElementByValue("input").should("have.attr", "value", option);
+        selectInput().should("have.attr", "aria-expanded", "false");
+        selectList().should("not.exist");
+      }
+    );
+
+    it.each([
+      ["A", "Amber", "Black", "Orange"],
+      ["O", "Brown", "Orange", "Yellow"],
+    ])(
+      "should filter options when %s is typed",
+      (text, optionValue1, optionValue2, optionValue3) => {
+        CypressMountWithProviders(<FilterableSelectComponent />);
+
+        commonDataElementInputPreview().type(text);
+        selectInput().should("have.attr", "aria-expanded", "true");
+        selectList().should("be.visible");
+        selectOption(positionOfElement("first"))
+          .should("have.text", optionValue1)
+          .and("be.visible")
+          .and("have.css", "background-color", "rgb(153, 173, 183)");
+        selectOption(positionOfElement("second"))
+          .should("have.text", optionValue2)
+          .and("be.visible");
```

Comment:
Why there is only background-colour checks only for optionValue 1, not optionValue2 or 3?

Reply:
I only checked that the first option had the background change applied as all the other options listed apart from the first one are as they are in a normal unfiltered list. I've added checks now, however, to confirm filtering has not altered the background of the other two options in the list.